rust/ommx/src/instance.rs

```diff
@@ -50,3 +50,285 @@ pub struct Instance {
     #[getset(get = "pub")]
     constraint_hints: ConstraintHints,
 }
+
+impl Instance {
+    ///
+    /// - All `VariableID`s in `Function`s contained both directly and indirectly must be keys of `decision_variables`.
+    /// - Key of `constraints` and `removed_constraints` are disjoint.
+    /// - The keys of `decision_variable_dependency` are also keys of `decision_variables`.
+    ///
+    #[allow(clippy::too_many_arguments)]
+    pub fn try_new(
+        sense: Sense,
+        objective: Function,
+        decision_variables: BTreeMap<VariableID, DecisionVariable>,
+        constraints: BTreeMap<ConstraintID, Constraint>,
+        removed_constraints: BTreeMap<ConstraintID, RemovedConstraint>,
+        decision_variable_dependency: BTreeMap<VariableID, Function>,
+        parameters: Option<v1::Parameters>,
+        description: Option<v1::instance::Description>,
+        constraint_hints: ConstraintHints,
+    ) -> anyhow::Result<Self> {
+        let instance = Instance {
+            sense,
+            objective,
+            decision_variables,
+            constraints,
+            removed_constraints,
+            decision_variable_dependency,
+            parameters,
+            description,
+            constraint_hints,
+        };
+
+        instance.validate()?;
```

Comment:
These validations are duplicated to `Parse` implementation. This will cause inconsistency in future. Use `Parse` implemetation or share its implementations in another way.